### PR TITLE
refactor: upgrade parachain-tokens to FRAME v2

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

### CODE REVIEW

File: parachain/runtime/src/lib.rs [hunk 1]
```diff
@@ -495,13 +500,107 @@ impl Convert<AccountId, [u8; 32]> for AccountId32Convert {
     }
 }
 
+#[cfg(feature = "cumulus-polkadot")]
```
Comment (sander2) [inline parachain/runtime/src/lib.rs:503]: Would it be possible to make an inline `mod` with this conditional compilation, and to put all items that now have individual compilation guards inside it?

File: crates/parachain-tokens/src/lib.rs [hunk 2]
```diff
@@ -4,120 +4,95 @@
...
+    #[pallet::pallet]
+    pub struct Pallet<T>(_);
 
-        /// Transfer issued tokens to parachain.
-        #[weight = 1000]
+    // The pallet's dispatchable functions.
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Transfer to parachain.
+        #[pallet::weight(1000)]
```
Comment (sander2) [inline crates/parachain-tokens/src/lib.rs:87]: We should probably make a benchmark for this, but happy to leave it for a future PR

Comment (gregdhill) [reply crates/parachain-tokens/src/lib.rs:87]: Yep, I'm somewhat inclined to wait for this as we may adopt some pallet from FRAME / cumulus / orml instead.

File: parachain/runtime/src/lib.rs (hunk 1 above)
Comment (gregdhill) [reply parachain/runtime/src/lib.rs:503]: Good idea, I will do this in a subsequent PR as I want to do the renaming of currency types soonish.

**Pull request merged**